bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairControllerHandler.java

```diff
@@ -0,0 +1,1186 @@
+
+                try {
+                    lightMode = PentairControllerConstants.LightMode.valueOf(str);
+                    setLightMode(lightMode);
+                } catch (IllegalArgumentException e) {
+                    logger.debug("Invalid light mode: {}", str);
+                }
+
+                break;
+            }
+            case CONTROLLER_SCHEDULESTRING: {
+                if (!(command instanceof StringType)) {
+                    break;
+                }
+
+                PentairControllerSchedule schedule = getScheduleByGroupID(groupId);
+
+                if (schedule == null) {
+                    break;
+                }
+
+                String str = ((StringType) command).toString();
+
+                if (!schedule.fromString(str)) {
+                    logger.debug("schedule invalid format: {}", str);
+                }
+
+                break;
+            }
+            case CONTROLLER_SCHEDULETYPE: {
+                if (!(command instanceof StringType)) {
+                    break;
+                }
+
+                PentairControllerSchedule schedule = getScheduleByGroupID(groupId);
+
+                if (schedule == null) {
+                    break;
+                }
+
+                String str = ((StringType) command).toString();
+                // save schedule only if a double update to the same value occurs within 5s of the last update
+                boolean bUpdate = (str.equals(schedule.getScheduleTypeStr())
+                        && ((System.currentTimeMillis() - lastScheduleTypeWrite) < 5000) && schedule.isDirty());
+                if (!schedule.setScheduleType(str)) {
+                    return;
+                }
+                lastScheduleTypeWrite = System.currentTimeMillis();
+
+                if (bUpdate) {
+                    saveSchedule(schedule);
+                    lastScheduleTypeWrite = 0;
+
+                    refreshScheduleChannels(channelUID, schedule);
+                }
+
+                break;
+            }
+            case CONTROLLER_SCHEDULESTART: {
+                if (!(command instanceof Number)) {
+                    break;
+                }
+
+                PentairControllerSchedule schedule = getScheduleByGroupID(groupId);
+
+                if (schedule == null) {
+                    break;
+                }
+
+                int start = ((Number) command).intValue();
+
+                schedule.setScheduleStart(start);
+
+                break;
+            }
+            case CONTROLLER_SCHEDULEEND: {
+                if (!(command instanceof Number)) {
+                    break;
+                }
+
+                PentairControllerSchedule schedule = getScheduleByGroupID(groupId);
+
+                if (schedule == null) {
+                    break;
+                }
+
+                int end = ((Number) command).intValue();
+
+                schedule.setScheduleEnd(end);
+
+                break;
```

Comment:
There are a lot of blank lines in the case statements, please remove most of them.